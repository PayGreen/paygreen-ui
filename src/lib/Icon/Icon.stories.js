import React from 'react';
import {
    PlaneIcon,
    OrganizationIcon,
    ShopIcon,
    TreesIcon,
    PointerIcon
} from './Icon';
import {
    colorPalletOptions,
    colorPalletDefault,
    colorThemeOptions,
    colorThemeDefault,
    greyOptions,
    greyDefault,
    formStatusOptions,
    formStatusDefault,
    iconSizeOptions,
    spaceOptions
} from '../../shared/constants';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, select, radios } from '@storybook/addon-knobs';

const colorPalletLabel = 'Color pallet';
const colorThemeLabel = 'Color theme';
const colorWabLabel = 'Grey color';
const colorStatusLabel = 'Status color';
const iconSizeLabel = 'Icon size';
const backgroundLabel = 'Has background';
const shadowLabel = 'With shadow';
const activeLabel = 'Active';
const marginTopLabel = 'Margin top';
const marginBottomLabel = 'Margin bottom';
const marginLeftLabel = 'Margin left';
const marginRightLabel = 'Margin right';
const centeredLabel = 'Centered';

storiesOf('Icons', module)
    .addDecorator(withKnobs)
    .add('Some icons', () => (
        <div>
            <PlaneIcon
                colorPallet={radios(colorPalletLabel, colorPalletOptions, colorPalletDefault)}
                colorTheme={select(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
                colorWab={select(colorWabLabel, greyOptions, greyDefault)}
                colorStatus={select(colorStatusLabel, formStatusOptions, formStatusDefault)}
                iconSize={select(iconSizeLabel, iconSizeOptions, iconSizeOptions.lg)}
                hasBackground={boolean(backgroundLabel, false)}
                shadowSize={boolean(shadowLabel, false)}
                isActive={boolean(activeLabel, false)}
                isCentered={boolean(centeredLabel, false)}
                marginTop={select(marginTopLabel, spaceOptions, spaceOptions.sm)}
                marginBottom={select(marginBottomLabel, spaceOptions, spaceOptions.sm)}
                marginLeft={select(marginLeftLabel, spaceOptions, spaceOptions.sm)}
                marginRight={select(marginRightLabel, spaceOptions, spaceOptions.sm)}
            />

            <OrganizationIcon
                colorPallet={radios(colorPalletLabel, colorPalletOptions, colorPalletDefault)}
                colorTheme={select(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
                colorWab={select(colorWabLabel, greyOptions, greyDefault)}
                colorStatus={select(colorStatusLabel, formStatusOptions, formStatusDefault)}
                iconSize={select(iconSizeLabel, iconSizeOptions, iconSizeOptions.lg)}
                hasBackground={boolean(backgroundLabel, false)}
                shadowSize={boolean(shadowLabel, false)}
                isActive={boolean(activeLabel, false)}
                isCentered={boolean(centeredLabel, false)}
                marginTop={select(marginTopLabel, spaceOptions, spaceOptions.sm)}
                marginBottom={select(marginBottomLabel, spaceOptions, spaceOptions.sm)}
                marginLeft={select(marginLeftLabel, spaceOptions, spaceOptions.sm)}
                marginRight={select(marginRightLabel, spaceOptions, spaceOptions.sm)}
            />

            <ShopIcon
                colorPallet={radios(colorPalletLabel, colorPalletOptions, colorPalletDefault)}
                colorTheme={select(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
                colorWab={select(colorWabLabel, greyOptions, greyDefault)}
                colorStatus={select(colorStatusLabel, formStatusOptions, formStatusDefault)}
                iconSize={select(iconSizeLabel, iconSizeOptions, iconSizeOptions.lg)}
                hasBackground={boolean(backgroundLabel, false)}
                shadowSize={boolean(shadowLabel, false)}
                isActive={boolean(activeLabel, false)}
                isCentered={boolean(centeredLabel, false)}
                marginTop={select(marginTopLabel, spaceOptions, spaceOptions.sm)}
                marginBottom={select(marginBottomLabel, spaceOptions, spaceOptions.sm)}
                marginLeft={select(marginLeftLabel, spaceOptions, spaceOptions.sm)}
                marginRight={select(marginRightLabel, spaceOptions, spaceOptions.sm)}
            />

            <TreesIcon
                colorPallet={radios(colorPalletLabel, colorPalletOptions, colorPalletDefault)}
                colorTheme={select(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
                colorWab={select(colorWabLabel, greyOptions, greyDefault)}
                colorStatus={select(colorStatusLabel, formStatusOptions, formStatusDefault)}
                iconSize={select(iconSizeLabel, iconSizeOptions, iconSizeOptions.lg)}
                hasBackground={boolean(backgroundLabel, false)}
                shadowSize={boolean(shadowLabel, false)}
                isActive={boolean(activeLabel, false)}
                isCentered={boolean(centeredLabel, false)}
                marginTop={select(marginTopLabel, spaceOptions, spaceOptions.sm)}
                marginBottom={select(marginBottomLabel, spaceOptions, spaceOptions.sm)}
                marginLeft={select(marginLeftLabel, spaceOptions, spaceOptions.sm)}
                marginRight={select(marginRightLabel, spaceOptions, spaceOptions.sm)}
            />

            <PointerIcon
                colorPallet={radios(colorPalletLabel, colorPalletOptions, colorPalletDefault)}
                colorTheme={select(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
                colorWab={select(colorWabLabel, greyOptions, greyDefault)}
                colorStatus={select(colorStatusLabel, formStatusOptions, formStatusDefault)}
                iconSize={select(iconSizeLabel, iconSizeOptions, iconSizeOptions.lg)}
                hasBackground={boolean(backgroundLabel, false)}
                shadowSize={boolean(shadowLabel, false)}
                isActive={boolean(activeLabel, false)}
                isCentered={boolean(centeredLabel, false)}
                marginTop={select(marginTopLabel, spaceOptions, spaceOptions.sm)}
                marginBottom={select(marginBottomLabel, spaceOptions, spaceOptions.sm)}
                marginLeft={select(marginLeftLabel, spaceOptions, spaceOptions.sm)}
                marginRight={select(marginRightLabel, spaceOptions, spaceOptions.sm)}
            />
        </div>
    ), {
        notes: 'Queen icons with some styles. You can add background, shadow, change colors, add active mode, center icon, change margins or icon size.',
    });