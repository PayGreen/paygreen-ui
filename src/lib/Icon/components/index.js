import ArrowBottom from './ArrowBottom';
import ArrowCircleLeft from './ArrowCircleLeft';
import ArrowCircleLeftBold from './ArrowCircleLeftBold';
import ArrowCircleRightBold from './ArrowCircleRightBold';
import ArrowCircleRight from './ArrowCircleRight';
import ArrowLeft from './ArrowLeft';
import ArrowLongBottom from './ArrowLongBottom';
import ArrowLongLeft from './ArrowLongLeft';
import ArrowLongRight from './ArrowLongRight';
import ArrowLongTop from './ArrowLongTop';
import ArrowRight from './ArrowRight';
import ArrowRoundBottomLeft from './ArrowRoundBottomLeft';
import ArrowRoundBottomRight from './ArrowRoundBottomRight';
import ArrowRoundTopLeft from './ArrowRoundTopLeft';
import ArrowRoundTopRight from './ArrowRoundTopRight';
import ArrowTop from './ArrowTop';
import BalloonBold from './BalloonBold';
import Balloon from './Balloon';
import Bank from './Bank';
import BeakerBold from './BeakerBold';
import Beaker from './Beaker';
import BellBold from './BellBold';
import Bell from './Bell';
import BulbBold from './BulbBold';
import Bulb from './Bulb';
import CardsBold from './CardsBold';
import Cards from './Cards';
import Car from './Car';
import Cart from './Cart';
import CautionBold from './CautionBold';
import Caution from './Caution';
import Change from './Change';
import CheckBold from './CheckBold';
import Checkbox from './Checkbox';
import Check from './Check';
import ClockBold from './ClockBold';
import Clock from './Clock';
import Code from './Code';
import Coins from './Coins';
import Contract from './Contract';
import CrossBold from './CrossBold';
import Cross from './Cross';
import Download from './Download';
import Empty from './Empty';
import Eye from './Eye';
import Factory from './Factory';
import Globe from './Globe';
import Heart from './Heart';
import Hourglass from './Hourglass';
import Information from './Information';
import Leaf from './Leaf';
import LetterBold from './LetterBold';
import Letter from './Letter';
import LinkBold from './LinkBold';
import Link from './Link';
import List from './List';
import Lock from './Lock';
import Marker from './Marker';
import Meal from './Meal';
import Megaphone from './Megaphone';
import MenuBold from './MenuBold';
import MenuDots from './MenuDots';
import Menu from './Menu';
import Meter from './Meter';
import Minus from './Minus';
import Moon from './Moon';
import Moove from './Moove';
import Organization from './Organization';
import Out from './Out';
import Paperclip from './Paperclip';
import Pen from './Pen';
import Phone from './Phone';
import Picture from './Picture';
import Plane from './Plane';
import Plus from './Plus';
import Pointer from './Pointer';
import Popin from './Popin';
import QuestionBold from './QuestionBold';
import Question from './Question';
import QuoteLeft from './QuoteLeft';
import QuoteRight from './QuoteRight';
import RefreshBold from './RefreshBold';
import Refresh from './Refresh';
import Satchel from './Satchel';
import Schedule from './Schedule';
import SearchBold from './SearchBold';
import Search from './Search';
import Shop from './Shop';
import Sun from './Sun';
import Support from './Support';
import Trees from './Trees';
import Trophy from './Trophy';
import Truck from './Truck';
import Tv from './Tv';
import UserBold from './UserBold';
import Users from './Users';
import User from './User';
import Wallet from './Wallet';
import Weight from './Weight';
import Wheels from './Wheels';

export {
    ArrowBottom,
    ArrowCircleLeft,
    ArrowCircleLeftBold,
    ArrowCircleRightBold,
    ArrowCircleRight,
    ArrowLeft,
    ArrowLongBottom,
    ArrowLongLeft,
    ArrowLongRight,
    ArrowLongTop,
    ArrowRight,
    ArrowRoundBottomLeft,
    ArrowRoundBottomRight,
    ArrowRoundTopLeft,
    ArrowRoundTopRight,
    ArrowTop,
    BalloonBold,
    Balloon,
    Bank,
    BeakerBold,
    Beaker,
    BellBold,
    Bell,
    BulbBold,
    Bulb,
    CardsBold,
    Cards,
    Car,
    Cart,
    CautionBold,
    Caution,
    Change,
    CheckBold,
    Checkbox,
    Check,
    ClockBold,
    Clock,
    Code,
    Coins,
    Contract,
    CrossBold,
    Cross,
    Download,
    Empty,
    Eye,
    Factory,
    Globe,
    Heart,
    Hourglass,
    Information,
    Leaf,
    LetterBold,
    Letter,
    LinkBold,
    Link,
    List,
    Lock,
    Marker,
    Meal,
    Megaphone,
    MenuBold,
    MenuDots,
    Menu,
    Meter,
    Minus,
    Moon,
    Moove,
    Organization,
    Out,
    Paperclip,
    Pen,
    Phone,
    Picture,
    Plane,
    Plus,
    Pointer,
    Popin,
    QuestionBold,
    Question,
    QuoteLeft,
    QuoteRight,
    RefreshBold,
    Refresh,
    Satchel,
    Schedule,
    SearchBold,
    Search,
    Shop,
    Sun,
    Support,
    Trees,
    Trophy,
    Truck,
    Tv,
    UserBold,
    Users,
    User,
    Wallet,
    Weight,
    Wheels,
};
