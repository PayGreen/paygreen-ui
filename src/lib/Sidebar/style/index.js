import styled from 'styled-components';
import { sidebarStyle } from './base';
import { TopbarBase } from '../../Topbar/style';
import { LogoBase } from '../../Logo/style';

const SidebarBase = styled.nav`
    ${sidebarStyle};
    .topbar-nav,
    .logo-nav {
        width: 100%;
    }

    @media (${props => props.theme.query.max.md}) {
        .topbar-nav {
            ${TopbarBase};
        }
        .logo-nav {
            display: none;
        }
    }

    @media (${props => props.theme.query.min.md}) {
        .topbar-nav {
            display: none;
        }
        .logo-nav {
            ${LogoBase};
            padding: ${props => props.theme.space.sm};
        }
    }
`;

export { SidebarBase };
