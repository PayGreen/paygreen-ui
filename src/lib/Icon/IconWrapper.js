import React from 'react';
import PropTypes from 'prop-types';
import {
    colorPalletOptions,
    colorPalletDefault,
    colorThemeOptions,
    colorThemeDefault,
    greyOptions,
    greyDefault,
    formStatusOptions,
    formStatusDefault,
    iconSizeOptions,
    spaceOptions,
    spaceDefault,
    iconHtmlTagOptions,
    iconHtmlTagDefault,
    rotateSizeOptions,
    rotateSizeDefault,
} from '../../shared/constants';
import { IconBase } from './style';

const maxDisplayed = 99;

const IconWrapper = ({ children, className, htmlTag, ...rest }) => {
    return (
        <IconBase
            {...rest}
            as={htmlTag}
            className={className ? 'icon ' + className : 'icon'}
        >
            {React.Children.map(children, child => {
                if (typeof child == 'object') {
                    return React.cloneElement(child);
                }
            })}

            {rest.number > 0 &&
            ![iconSizeOptions.xs, iconSizeOptions.xxs].includes(
                rest.iconSize,
            ) ? (
                <span className="badge-number">
                    {rest.number <= maxDisplayed
                        ? rest.number
                        : maxDisplayed + '+'}
                </span>
            ) : null}
        </IconBase>
    );
};

IconWrapper.propTypes = {
    htmlTag: PropTypes.oneOf(Object.values(iconHtmlTagOptions)),
    number: PropTypes.number,
    colorPallet: PropTypes.oneOf(Object.values(colorPalletOptions)),
    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    colorWab: PropTypes.oneOf(Object.values(greyOptions)),
    colorStatus: PropTypes.oneOf(Object.values(formStatusOptions)),
    iconSize: PropTypes.oneOf(Object.values(iconSizeOptions)),
    hasBackground: PropTypes.bool,
    hasHover: PropTypes.bool,
    isActive: PropTypes.bool,
    isCentered: PropTypes.bool,
    marginTop: PropTypes.oneOf(Object.values(spaceOptions)),
    marginBottom: PropTypes.oneOf(Object.values(spaceOptions)),
    marginLeft: PropTypes.oneOf(Object.values(spaceOptions)),
    marginRight: PropTypes.oneOf(Object.values(spaceOptions)),
    rotateSize: PropTypes.oneOf(Object.values(rotateSizeOptions)),
};

IconWrapper.defaultProps = {
    htmlTag: iconHtmlTagDefault,
    number: 0,
    colorPallet: colorPalletDefault,
    colorTheme: colorThemeDefault,
    colorWab: greyDefault,
    colorStatus: formStatusDefault,
    iconSize: iconSizeOptions.sm,
    hasBackground: false,
    hasHover: false,
    isActive: false,
    isCentered: false,
    marginTop: spaceDefault,
    marginBottom: spaceDefault,
    marginLeft: spaceDefault,
    marginRight: spaceDefault,
    rotateSize: rotateSizeDefault,
};

export default IconWrapper;
