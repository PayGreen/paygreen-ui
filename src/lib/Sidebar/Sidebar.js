import React from 'react';
import { SidebarBase } from './style';

const Sidebar = props => {
    return <SidebarBase {...props}>{props.children}</SidebarBase>;
};

export default Sidebar;