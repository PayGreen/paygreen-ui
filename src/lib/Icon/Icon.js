import React from 'react';
import IconWrapper from './IconWrapper';
import {
  ArrowBottom,
  ArrowCircleLeft,
  ArrowCircleLeftBold,
  ArrowCircleRightBold,
  ArrowCircleRight,
  ArrowLeft,
  ArrowLongBottom,
  ArrowLongLeft,
  ArrowLongRight,
  ArrowLongTop,
  ArrowRight,
  ArrowRoundBottomLeft,
  ArrowRoundBottomRight,
  ArrowRoundTopLeft,
  ArrowRoundTopRight,
  ArrowTop,
  BalloonBold,
  Balloon,
  Bank,
  BeakerBold,
  Beaker,
  BellBold,
  Bell,
  BulbBold,
  Bulb,
  CardsBold,
  Cards,
  Car,
  Cart,
  CautionBold,
  Caution,
  Change,
  CheckBold,
  Checkbox,
  Check,
  ClockBold,
  Clock,
  Code,
  Coins,
  Contract,
  CrossBold,
  Cross,
  Download,
  Empty,
  Eye,
  Factory,
  Globe,
  Heart,
  Hourglass,
  Information,
  Leaf,
  LetterBold,
  Letter,
  LinkBold,
  Link,
  List,
  Lock,
  Marker,
  Meal,
  Megaphone,
  MenuBold,
  MenuDots,
  Menu,
  Meter,
  Minus,
  Moon,
  Moove,
  Organization,
  Out,
  Paperclip,
  Pen,
  Phone,
  Picture,
  Plane,
  Plus,
  Pointer,
  Popin,
  QuestionBold,
  Question,
  QuoteLeft,
  QuoteRight,
  RefreshBold,
  Refresh,
  Satchel,
  Schedule,
  SearchBold,
  Search,
  Shop,
  Sun,
  Support,
  Trees,
  Trophy,
  Truck,
  Tv,
  UserBold,
  Users,
  User,
  Wallet,
  Weight,
  Wheels,
} from './components';

export const ArrowBottomIcon = (props) => <IconWrapper {...props}><ArrowBottom /></IconWrapper>
export const ArrowCircleLeftIcon = (props) => <IconWrapper {...props}><ArrowCircleLeft /></IconWrapper>
export const ArrowCircleLeftBoldIcon = (props) => <IconWrapper {...props}><ArrowCircleLeftBold /></IconWrapper>
export const ArrowCircleRightBoldIcon = (props) => <IconWrapper {...props}><ArrowCircleRightBold /></IconWrapper>
export const ArrowCircleRightIcon = (props) => <IconWrapper {...props}><ArrowCircleRight /></IconWrapper>
export const ArrowLeftIcon = (props) => <IconWrapper {...props}><ArrowLeft /></IconWrapper>
export const ArrowLongBottomIcon = (props) => <IconWrapper {...props}><ArrowLongBottom /></IconWrapper>
export const ArrowLongLeftIcon = (props) => <IconWrapper {...props}><ArrowLongLeft /></IconWrapper>
export const ArrowLongRightIcon = (props) => <IconWrapper {...props}><ArrowLongRight /></IconWrapper>
export const ArrowLongTopIcon = (props) => <IconWrapper {...props}><ArrowLongTop /></IconWrapper>
export const ArrowRightIcon = (props) => <IconWrapper {...props}><ArrowRight /></IconWrapper>
export const ArrowRoundBottomLeftIcon = (props) => <IconWrapper {...props}><ArrowRoundBottomLeft /></IconWrapper>
export const ArrowRoundBottomRightIcon = (props) => <IconWrapper {...props}><ArrowRoundBottomRight /></IconWrapper>
export const ArrowRoundTopLeftIcon = (props) => <IconWrapper {...props}><ArrowRoundTopLeft /></IconWrapper>
export const ArrowRoundTopRightIcon = (props) => <IconWrapper {...props}><ArrowRoundTopRight /></IconWrapper>
export const ArrowTopIcon = (props) => <IconWrapper {...props}><ArrowTop /></IconWrapper>
export const BalloonBoldIcon = (props) => <IconWrapper {...props}><BalloonBold /></IconWrapper>
export const BalloonIcon = (props) => <IconWrapper {...props}><Balloon /></IconWrapper>
export const BankIcon = (props) => <IconWrapper {...props}><Bank /></IconWrapper>
export const BeakerBoldIcon = (props) => <IconWrapper {...props}><BeakerBold /></IconWrapper>
export const BeakerIcon = (props) => <IconWrapper {...props}><Beaker /></IconWrapper>
export const BellBoldIcon = (props) => <IconWrapper {...props}><BellBold /></IconWrapper>
export const BellIcon = (props) => <IconWrapper {...props}><Bell /></IconWrapper>
export const BulbBoldIcon = (props) => <IconWrapper {...props}><BulbBold /></IconWrapper>
export const BulbIcon = (props) => <IconWrapper {...props}><Bulb /></IconWrapper>
export const CardsBoldIcon = (props) => <IconWrapper {...props}><CardsBold /></IconWrapper>
export const CardsIcon = (props) => <IconWrapper {...props}><Cards /></IconWrapper>
export const CarIcon = (props) => <IconWrapper {...props}><Car /></IconWrapper>
export const CartIcon = (props) => <IconWrapper {...props}><Cart /></IconWrapper>
export const CautionBoldIcon = (props) => <IconWrapper {...props}><CautionBold /></IconWrapper>
export const CautionIcon = (props) => <IconWrapper {...props}><Caution /></IconWrapper>
export const ChangeIcon = (props) => <IconWrapper {...props}><Change /></IconWrapper>
export const CheckBoldIcon = (props) => <IconWrapper {...props}><CheckBold /></IconWrapper>
export const CheckboxIcon = (props) => <IconWrapper {...props}><Checkbox /></IconWrapper>
export const CheckIcon = (props) => <IconWrapper {...props}><Check /></IconWrapper>
export const ClockBoldIcon = (props) => <IconWrapper {...props}><ClockBold /></IconWrapper>
export const ClockIcon = (props) => <IconWrapper {...props}><Clock /></IconWrapper>
export const CodeIcon = (props) => <IconWrapper {...props}><Code /></IconWrapper>
export const CoinsIcon = (props) => <IconWrapper {...props}><Coins /></IconWrapper>
export const ContractIcon = (props) => <IconWrapper {...props}><Contract /></IconWrapper>
export const CrossBoldIcon = (props) => <IconWrapper {...props}><CrossBold /></IconWrapper>
export const CrossIcon = (props) => <IconWrapper {...props}><Cross /></IconWrapper>
export const DownloadIcon = (props) => <IconWrapper {...props}><Download /></IconWrapper>
export const EmptyIcon = (props) => <IconWrapper {...props}><Empty /></IconWrapper>
export const EyeIcon = (props) => <IconWrapper {...props}><Eye /></IconWrapper>
export const FactoryIcon = (props) => <IconWrapper {...props}><Factory /></IconWrapper>
export const GlobeIcon = (props) => <IconWrapper {...props}><Globe /></IconWrapper>
export const HeartIcon = (props) => <IconWrapper {...props}><Heart /></IconWrapper>
export const HourglassIcon = (props) => <IconWrapper {...props}><Hourglass /></IconWrapper>
export const InformationIcon = (props) => <IconWrapper {...props}><Information /></IconWrapper>
export const LeafIcon = (props) => <IconWrapper {...props}><Leaf /></IconWrapper>
export const LetterBoldIcon = (props) => <IconWrapper {...props}><LetterBold /></IconWrapper>
export const LetterIcon = (props) => <IconWrapper {...props}><Letter /></IconWrapper>
export const LinkBoldIcon = (props) => <IconWrapper {...props}><LinkBold /></IconWrapper>
export const LinkIcon = (props) => <IconWrapper {...props}><Link /></IconWrapper>
export const ListIcon = (props) => <IconWrapper {...props}><List /></IconWrapper>
export const LockIcon = (props) => <IconWrapper {...props}><Lock /></IconWrapper>
export const MarkerIcon = (props) => <IconWrapper {...props}><Marker /></IconWrapper>
export const MealIcon = (props) => <IconWrapper {...props}><Meal /></IconWrapper>
export const MegaphoneIcon = (props) => <IconWrapper {...props}><Megaphone /></IconWrapper>
export const MenuBoldIcon = (props) => <IconWrapper {...props}><MenuBold /></IconWrapper>
export const MenuDotsIcon = (props) => <IconWrapper {...props}><MenuDots /></IconWrapper>
export const MenuIcon = (props) => <IconWrapper {...props}><Menu /></IconWrapper>
export const MeterIcon = (props) => <IconWrapper {...props}><Meter /></IconWrapper>
export const MinusIcon = (props) => <IconWrapper {...props}><Minus /></IconWrapper>
export const MoonIcon = (props) => <IconWrapper {...props}><Moon /></IconWrapper>
export const MooveIcon = (props) => <IconWrapper {...props}><Moove /></IconWrapper>
export const OrganizationIcon = (props) => <IconWrapper {...props}><Organization /></IconWrapper>
export const OutIcon = (props) => <IconWrapper {...props}><Out /></IconWrapper>
export const PaperclipIcon = (props) => <IconWrapper {...props}><Paperclip /></IconWrapper>
export const PenIcon = (props) => <IconWrapper {...props}><Pen /></IconWrapper>
export const PhoneIcon = (props) => <IconWrapper {...props}><Phone /></IconWrapper>
export const PictureIcon = (props) => <IconWrapper {...props}><Picture /></IconWrapper>
export const PlaneIcon = (props) => <IconWrapper {...props}><Plane /></IconWrapper>
export const PlusIcon = (props) => <IconWrapper {...props}><Plus /></IconWrapper>
export const PointerIcon = (props) => <IconWrapper {...props}><Pointer /></IconWrapper>
export const PopinIcon = (props) => <IconWrapper {...props}><Popin /></IconWrapper>
export const QuestionBoldIcon = (props) => <IconWrapper {...props}><QuestionBold /></IconWrapper>
export const QuestionIcon = (props) => <IconWrapper {...props}><Question /></IconWrapper>
export const QuoteLeftIcon = (props) => <IconWrapper {...props}><QuoteLeft /></IconWrapper>
export const QuoteRightIcon = (props) => <IconWrapper {...props}><QuoteRight /></IconWrapper>
export const RefreshBoldIcon = (props) => <IconWrapper {...props}><RefreshBold /></IconWrapper>
export const RefreshIcon = (props) => <IconWrapper {...props}><Refresh /></IconWrapper>
export const SatchelIcon = (props) => <IconWrapper {...props}><Satchel /></IconWrapper>
export const ScheduleIcon = (props) => <IconWrapper {...props}><Schedule /></IconWrapper>
export const SearchBoldIcon = (props) => <IconWrapper {...props}><SearchBold /></IconWrapper>
export const SearchIcon = (props) => <IconWrapper {...props}><Search /></IconWrapper>
export const ShopIcon = (props) => <IconWrapper {...props}><Shop /></IconWrapper>
export const SunIcon = (props) => <IconWrapper {...props}><Sun /></IconWrapper>
export const SupportIcon = (props) => <IconWrapper {...props}><Support /></IconWrapper>
export const TreesIcon = (props) => <IconWrapper {...props}><Trees /></IconWrapper>
export const TrophyIcon = (props) => <IconWrapper {...props}><Trophy /></IconWrapper>
export const TruckIcon = (props) => <IconWrapper {...props}><Truck /></IconWrapper>
export const TvIcon = (props) => <IconWrapper {...props}><Tv /></IconWrapper>
export const UserBoldIcon = (props) => <IconWrapper {...props}><UserBold /></IconWrapper>
export const UsersIcon = (props) => <IconWrapper {...props}><Users /></IconWrapper>
export const UserIcon = (props) => <IconWrapper {...props}><User /></IconWrapper>
export const WalletIcon = (props) => <IconWrapper {...props}><Wallet /></IconWrapper>
export const WeightIcon = (props) => <IconWrapper {...props}><Weight /></IconWrapper>
export const WheelsIcon = (props) => <IconWrapper {...props}><Wheels /></IconWrapper>
