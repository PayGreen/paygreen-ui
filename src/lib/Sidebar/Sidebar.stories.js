import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, radios, withKnobs } from '@storybook/addon-knobs';
import {
    folder,
    colorPalletOptions,
    colorThemeOptions,
    colorThemeDefault,
    greyOptions,
    spaceOptions,
    iconSizeOptions,
    iconHtmlTagOptions,
    sidebarItemHtmlTagOptions,
    hoverDirectionOptions,
    alignDefault,
} from '../../shared/constants';
import labels from '../../shared/labels';
import { ArrowRightIcon, LeafIcon, MeterIcon } from '../Icon/Icon';
import { logo } from '../Logo/sample/logo';
import Logo from '../Logo/Logo';
import Topbar from '../Topbar/Topbar';
import Link from '../Link/Link';
import Box from '../Box/Box';
import SidebarMenuCategory from '../SidebarMenuCategory/SidebarMenuCategory';
import SidebarItem from '../SidebarItem/SidebarItem';
import SidebarList from '../SidebarList/SidebarList';
import SidebarMenu from '../SidebarMenu/SidebarMenu';
import Sidebar from './Sidebar';

const logoSidebar = (
    <Logo
        align={alignDefault}
        hasThemeColor={true}
        hasBaseline={false}
        hoverDirection={hoverDirectionOptions.right}
        blockWidth={spaceOptions.sm}
    >
        {logo}
    </Logo>
);

storiesOf(folder.nav + 'Sidebar', module)
    .addDecorator(withKnobs)
    .add('Sidebar', () => (
        <Sidebar>
            <Topbar className="hideOnDesktop">
                <a href="#">{logoSidebar}</a>

                <ArrowRightIcon
                    htmlTag={iconHtmlTagOptions.button}
                    iconSize={iconSizeOptions.xl}
                    colorPallet={colorPalletOptions.wab}
                    colorWab={greyOptions.grey40}
                />
            </Topbar>

            <Box
                className="hideOnTablet"
                marginLeft={spaceOptions.md}
                marginTop={spaceOptions.md}
                marginBottom={spaceOptions.md}
            >
                <a href="#">{logoSidebar}</a>
            </Box>

            <Box marginTop={spaceOptions.md}>
                <SidebarMenuCategory
                    categoryTitle="CatÃ©gorie de Pages"
                    paddingLateral={spaceOptions.md}
                >
                    <SidebarMenu
                        colorTheme={radios(
                            labels.colorTheme,
                            colorThemeOptions,
                            colorThemeDefault,
                        )}
                        paddingLateral={spaceOptions.md}
                    >
                        <SidebarItem
                            htmlTag={sidebarItemHtmlTagOptions.button}
                            colorTheme={radios(
                                labels.colorTheme,
                                colorThemeOptions,
                                colorThemeDefault,
                            )}
                            paddingLateral={spaceOptions.md}
                        >
                            <MeterIcon
                                iconSize={iconSizeOptions.lg}
                                colorTheme={radios(
                                    labels.colorTheme,
                                    colorThemeOptions,
                                    colorThemeDefault,
                                )}
                            />

                            <Link
                                colorTheme={radios(
                                    labels.colorTheme,
                                    colorThemeOptions,
                                    colorThemeDefault,
                                )}
                            >
                                Tableau de Bord
                            </Link>
                        </SidebarItem>

                        <SidebarList>
                            <a href="#">
                                <SidebarItem
                                    colorTheme={radios(
                                        labels.colorTheme,
                                        colorThemeOptions,
                                        colorThemeDefault,
                                    )}
                                    paddingLateral={spaceOptions.md}
                                >
                                    <Link
                                        colorPallet={colorPalletOptions.wab}
                                        colorWab={greyOptions.grey60}
                                    >
                                        Vue d'ensemble
                                    </Link>
                                </SidebarItem>
                            </a>

                            <a href="#">
                                <SidebarItem
                                    colorTheme={radios(
                                        labels.colorTheme,
                                        colorThemeOptions,
                                        colorThemeDefault,
                                    )}
                                    paddingLateral={spaceOptions.md}
                                >
                                    <Link
                                        colorPallet={colorPalletOptions.wab}
                                        colorWab={greyOptions.grey60}
                                    >
                                        Vue par projet
                                    </Link>
                                </SidebarItem>
                            </a>
                        </SidebarList>
                    </SidebarMenu>

                    <SidebarMenu
                        colorTheme={radios(
                            labels.colorTheme,
                            colorThemeOptions,
                            colorThemeDefault,
                        )}
                        paddingLateral={spaceOptions.md}
                    >
                        <a href="#">
                            <SidebarItem
                                colorTheme={radios(
                                    labels.colorTheme,
                                    colorThemeOptions,
                                    colorThemeDefault,
                                )}
                                paddingLateral={spaceOptions.md}
                                isActive={boolean('Is Active', false)}
                            >
                                <LeafIcon
                                    iconSize={iconSizeOptions.lg}
                                    colorTheme={radios(
                                        labels.colorTheme,
                                        colorThemeOptions,
                                        colorThemeDefault,
                                    )}
                                />

                                <Link
                                    colorTheme={radios(
                                        labels.colorTheme,
                                        colorThemeOptions,
                                        colorThemeDefault,
                                    )}
                                >
                                    Compensations
                                </Link>
                            </SidebarItem>
                        </a>
                    </SidebarMenu>
                </SidebarMenuCategory>
            </Box>
        </Sidebar>
    ));
