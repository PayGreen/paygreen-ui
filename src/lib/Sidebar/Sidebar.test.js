import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import Sidebar from './Sidebar';
import {
    colorPalletOptions,
    greyOptions,
    iconSizeOptions,
} from '../../shared/constants';
import { ArrowRightIcon } from '../Icon/Icon';

it('renders without crashing', () => {
    const sidebar = TestRenderer.create(
        <Sidebar theme={ThemeDefault}>
            <ArrowRightIcon theme={ThemeDefault} />
        </Sidebar>,
    );
    expect(sidebar.toJSON()).toMatchSnapshot();
});
