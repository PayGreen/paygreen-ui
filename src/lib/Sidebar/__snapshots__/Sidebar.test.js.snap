// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="style__SidebarBase-qd29dm-0 htbAsZ"
>
  <span
    className="style__IconBase-ak3679-0 cZjoZu icon"
  >
    <svg
      viewBox="0 0 143 143"
    >
      <path
        d="M43.5 123.9c-1 0-1.9-.4-2.6-1.2-1.3-1.5-1.1-3.7.3-4.9l52-45.6-52-45c-1.5-1.3-1.6-3.5-.4-4.9 1.3-1.5 3.5-1.6 4.9-.4l55 47.6c.8.7 1.2 1.6 1.2 2.6s-.4 2-1.2 2.6l-55 48.2c-.6.7-1.4 1-2.2 1z"
      />
    </svg>
  </span>
</div>
`;
